remove name datatype data migration scripts and non_upgradeable tests

kalensk opened the pull request:
The name data type is supported for upgrades. See: 
https://github.com/greenplum-db/gpdb/pull/15114
https://github.com/greenplum-db/gpupgrade/pull/765

Note: Public facing docs will need to be updated as well as postgis "fixup" scripts which are no longer needed and associated upgrade docs.

Pipeline: https://cm.ci.gpdb.pivotal.io/teams/main/pipelines/gpupgrade:dataMigrationNameDatatype
Note: Pipeline is red until https://github.com/greenplum-db/gpdb/pull/15114 is merged as the jobs are hitting that pg_upgrade check.

jimmyyih reviewed (commented):
The draft looks fine to me assuming the dev pipeline goes green.  There are some references left still though if they should be changed too:
```
2023-03-07 18:53:39 ☕  jyihRJGH5 in ~/workspace/gpupgrade/data-migration-scripts/5-to-6-seed-scripts
± → grep -Iinr "both name and tsquery" *
finalize/tables_using_tsquery_type/recreate_views_on_deprecated_built_in_types.sql:4:-- Combine both name and tsquery scripts into the same subdirectory since both
finalize/tables_using_tsquery_type/recreate_indexes_on_deprecated_built_in_types.sql:4:-- Combine both name and tsquery scripts into the same subdirectory since both
initialize/tables_using_tsquery_type/gen_drop_depr_built_in_type_dependent_views.sql:4:-- Combine both name and tsquery scripts into the same subdirectory since both
revert/tables_using_tsquery_type/recreate_views_on_deprecated_built_in_types.sql:4:-- Combine both name and tsquery scripts into the same subdirectory since both
revert/tables_using_tsquery_type/recreate_indexes_on_deprecated_built_in_types.sql:4:-- Combine both name and tsquery scripts into the same subdirectory since both
test/create_nonupgradable_objects.sql:184:-- view on both name and tsquery from the same table
test/create_nonupgradable_objects.sql:188:-- view on both name and tsquery from multiple tables
```

kalensk commented:
> There are some references left still though if they should be changed too:

Thanks! 

1. I removed or updated those references you listed.
2. Given what you mentioned I took another closer look at `5-to-6-seed-scripts/test/create_nonupgradable_objects.sql` and `6-to-7-seed-scripts/test/create_nonupgradable_objects.sql` (they are basically the same at the moment).

I removed a few of the tests on name data type and would like to ensure we have similar coverage in the pg_upgrade acceptance tests for name https://github.com/greenplum-db/gpupgrade/pull/765 

Some of the tests include:
- inheriting a table contianing a name data type
- partition table containing name data type
- table with index on the name data typ
- "view on a view on a name column with owner different than the underlying table"
- Take a closer look as I might have missed some as things get a bit complicated and tricky.


3. This part does not make sense to me since I am not seeing where the underlying table is that is actually testing these views. Any help in understanding the below? I know there were tricky issues (can't remember all the details at this moment) in which doing this level of testing for data types and depreciated data types used within views is necessary. Can we be sure we have that coverage in the pg_upgrade acceptance tests for name https://github.com/greenplum-db/gpupgrade/pull/765?
```
-- view on a view on a name column with owner different than the underlying table
DROP VIEW IF EXISTS v3_on_v2_recursive;
CREATE VIEW v3_on_v2_recursive AS SELECT * FROM v2_on_t2_with_name;
ALTER TABLE v3_on_v2_recursive OWNER TO test_role2;

-- Third level recursive view on a name column
DROP VIEW IF EXISTS v4_on_v3_recursive;
CREATE VIEW v4_on_v3_recursive AS SELECT * FROM v3_on_v2_recursive;

-- view on a table with view that doesn't actually depend on the name column
-- this one should not be dropped since its dependent columns are not changing
DROP VIEW IF EXISTS v4_on_t2_no_name;
CREATE VIEW v4_on_t2_no_name AS SELECT a, 1::INTEGER AS i FROM t2_with_name;
```

jimmyyih commented:
> Some of the tests include:
>    inheriting a table containing a name data type
>    partition table containing name data type
>    table with index on the name data type
>    "view on a view on a name column with owner different than the underlying table"
>    Take a closer look as I might have missed some as things get a bit complicated and tricky.

I replied for most of these in https://github.com/greenplum-db/gpupgrade/pull/765.  The only one that makes sense to preserve would be the index on a name column.  I'll add that in really quick.

> This part does not make sense to me since I am not seeing where the underlying table is that is actually testing these views. Any help in understanding the below?

The v3_on_v2_recursive view is dependent on the v2_on_t2_with_name view.  The v2_on_t2_with_name view is dependent on t2_with_name table which is removed in this PR already.
```
CREATE TABLE t2_with_name(a int, b name) DISTRIBUTED RANDOMLY;
INSERT INTO t2_with_name SELECT 1, 'bbb';
CREATE VIEW v2_on_t2_with_name AS SELECT * FROM t2_with_name;
ALTER TABLE v2_on_t2_with_name OWNER TO test_role1;
```

That said, the block of SQL you highlighted can be removed.

jimmyyih commented on data-migration-scripts/5-to-6-seed-scripts/test/create_nonupgradable_objects.sql:
```diff
@@ -221,23 +161,13 @@ CREATE VIEW v4_on_v3_recursive AS SELECT * FROM v3_on_v2_recursive;
-CREATE INDEX sales_name_idx on sales_name(office_name);
+-- view on tsquery from multiple tables
+DROP VIEW IF EXISTS view_on_tsquery_mult_tables;
+CREATE VIEW view_on_tsquery_mult_tables AS SELECT t1.population, t2.altitude FROM table_with_tsquery t1, table_with_tsquery t2;
```
The `table_with_tsquery` table doesn't have a `population` column anymore.  Can just change it to `t1.name`.

kalensk commented:
> I replied for most of these in https://github.com/greenplum-db/gpupgrade/pull/765. The only one that makes sense to preserve would be the index on a name column. I'll add that in really quick.

Excellent. Glad we got "full" coverage.

> The v3_on_v2_recursive view is dependent on the v2_on_t2_with_name view. The v2_on_t2_with_name view is dependent on t2_with_name table which is removed in this PR already.

Ah, I missed that `t2_with_name` was already removed at first. Makes sense. Thank you for clarifying.

> That said, the block of SQL you highlighted can be removed.

Will update.

kalensk commented:
No longer dropping the postgis view's containing name data type results in the source and cluster dumps to be different.

Need to add a new filter for the following ARRAY ::text casting:

source 5X dump:
```SQL
AND ((c.relkind)::text = ANY ((ARRAY['r'::bpchar::character(1), 'v'::bpchar::character(1), 'm'::bpchar::character(1), 'f'::bpchar::character(1)])::text[]))) 
```

target 6X dump:
```SQL
AND ((c.relkind)::text = ANY (ARRAY[('r'::character(1))::text, ('v'::character(1))::text, ('m'::character(1))::text, ('f'::character(1))::text]))) 
```

More complete diff:
```diff
--- /tmp/source.sql.filtered	2023-03-10 03:15:36.532083771 +0000
+++ /tmp/target.sql.filtered	2023-03-10 03:15:21.043665813 +0000

 
 CREATE VIEW public.raster_columns AS
-SELECT current_database() AS r_table_catalog, n.nspname AS r_table_schema, c.relname AS r_table_name, a.attname AS r_raster_column, COALESCE(public._raster_constraint_info_srid(n.nspname, c.relname, a.attname), (SELECT public.st_srid('010100000000000000000000000000000000000000'::public.geometry) AS st_srid)) AS srid, public._raster_constraint_info_scale(n.nspname, c.relname, a.attname, 'x'::bpchar) AS scale_x, public._raster_constraint_info_scale(n.nspname, c.relname, a.attname, 'y'::bpchar) AS scale_y, public._raster_constraint_info_blocksize(n.nspname, c.relname, a.attname, 'width'::text) AS blocksize_x, public._raster_constraint_info_blocksize(n.nspname, c.relname, a.attname, 'height'::text) AS blocksize_y, COALESCE(public._raster_constraint_info_alignment(n.nspname, c.relname, a.attname), false) AS same_alignment, COALESCE(public._raster_constraint_info_regular_blocking(n.nspname, c.relname, a.attname), false) AS regular_blocking, public._raster_constraint_info_num_bands(n.nspname, c.relname, a.attname) AS num_bands, public._raster_constraint_info_pixel_types(n.nspname, c.relname, a.attname) AS pixel_types, public._raster_constraint_info_nodata_values(n.nspname, c.relname, a.attname) AS nodata_values, public._raster_constraint_info_out_db(n.nspname, c.relname, a.attname) AS out_db, public._raster_constraint_info_extent(n.nspname, c.relname, a.attname) AS extent FROM pg_class c, pg_attribute a, pg_type t, pg_namespace n WHERE (((((((t.typname = 'raster'::name) AND (a.attisdropped = false)) AND (a.atttypid = t.oid)) AND (a.attrelid = c.oid)) AND (c.relnamespace = n.oid)) AND ((c.relkind)::text = ANY ((ARRAY['r'::bpchar::character(1), 'v'::bpchar::character(1), 'm'::bpchar::character(1), 'f'::bpchar::character(1)])::text[]))) AND (NOT pg_is_other_temp_schema(c.relnamespace)));
+SELECT current_database() AS r_table_catalog, n.nspname AS r_table_schema, c.relname AS r_table_name, a.attname AS r_raster_column, COALESCE(public._raster_constraint_info_srid(n.nspname, c.relname, a.attname), (SELECT public.st_srid('010100000000000000000000000000000000000000'::public.geometry) AS st_srid)) AS srid, public._raster_constraint_info_scale(n.nspname, c.relname, a.attname, 'x'::bpchar) AS scale_x, public._raster_constraint_info_scale(n.nspname, c.relname, a.attname, 'y'::bpchar) AS scale_y, public._raster_constraint_info_blocksize(n.nspname, c.relname, a.attname, 'width'::text) AS blocksize_x, public._raster_constraint_info_blocksize(n.nspname, c.relname, a.attname, 'height'::text) AS blocksize_y, COALESCE(public._raster_constraint_info_alignment(n.nspname, c.relname, a.attname), false) AS same_alignment, COALESCE(public._raster_constraint_info_regular_blocking(n.nspname, c.relname, a.attname), false) AS regular_blocking, public._raster_constraint_info_num_bands(n.nspname, c.relname, a.attname) AS num_bands, public._raster_constraint_info_pixel_types(n.nspname, c.relname, a.attname) AS pixel_types, public._raster_constraint_info_nodata_values(n.nspname, c.relname, a.attname) AS nodata_values, public._raster_constraint_info_out_db(n.nspname, c.relname, a.attname) AS out_db, public._raster_constraint_info_extent(n.nspname, c.relname, a.attname) AS extent FROM pg_class c, pg_attribute a, pg_type t, pg_namespace n WHERE (((((((t.typname = 'raster'::name) AND (a.attisdropped = false)) AND (a.atttypid = t.oid)) AND (a.attrelid = c.oid)) AND (c.relnamespace = n.oid)) AND ((c.relkind)::text = ANY (ARRAY[('r'::character(1))::text, ('v'::character(1))::text, ('m'::character(1))::text, ('f'::character(1))::text]))) AND (NOT pg_is_other_temp_schema(c.relnamespace)));
 
 
 ALTER TABLE public.raster_columns OWNER TO gpadmin;
@@ -62068,7 +62069,7 @@
 --
 
 CREATE VIEW public.raster_overviews AS
-SELECT current_database() AS o_table_catalog, n.nspname AS o_table_schema, c.relname AS o_table_name, a.attname AS o_raster_column, current_database() AS r_table_catalog, (split_part(split_part(s.consrc, '''::name'::text, 1), ''''::text, 2))::name AS r_table_schema, (split_part(split_part(s.consrc, '''::name'::text, 2), ''''::text, 2))::name AS r_table_name, (split_part(split_part(s.consrc, '''::name'::text, 3), ''''::text, 2))::name AS r_raster_column, (btrim(split_part(s.consrc, ','::text, 2)))::integer AS overview_factor FROM pg_class c, pg_attribute a, pg_type t, pg_namespace n, pg_constraint s WHERE ((((((((((t.typname = 'raster'::name) AND (a.attisdropped = false)) AND (a.atttypid = t.oid)) AND (a.attrelid = c.oid)) AND (c.relnamespace = n.oid)) AND ((c.relkind)::text = ANY ((ARRAY['r'::bpchar::character(1), 'v'::bpchar::character(1), 'm'::bpchar::character(1), 'f'::bpchar::character(1)])::text[]))) AND (s.connamespace = n.oid)) AND (s.conrelid = c.oid)) AND (s.consrc ~~ '%_overview_constraint(%'::text)) AND (NOT pg_is_other_temp_schema(c.relnamespace)));
+SELECT current_database() AS o_table_catalog, n.nspname AS o_table_schema, c.relname AS o_table_name, a.attname AS o_raster_column, current_database() AS r_table_catalog, (split_part(split_part(s.consrc, '''::name'::text, 1), ''''::text, 2))::name AS r_table_schema, (split_part(split_part(s.consrc, '''::name'::text, 2), ''''::text, 2))::name AS r_table_name, (split_part(split_part(s.consrc, '''::name'::text, 3), ''''::text, 2))::name AS r_raster_column, (btrim(split_part(s.consrc, ','::text, 2)))::integer AS overview_factor FROM pg_class c, pg_attribute a, pg_type t, pg_namespace n, pg_constraint s WHERE ((((((((((t.typname = 'raster'::name) AND (a.attisdropped = false)) AND (a.atttypid = t.oid)) AND (a.attrelid = c.oid)) AND (c.relnamespace = n.oid)) AND ((c.relkind)::text = ANY (ARRAY[('r'::character(1))::text, ('v'::character(1))::text, ('m'::character(1))::text, ('f'::character(1))::text]))) AND (s.connamespace = n.oid)) AND (s.conrelid = c.oid)) AND (s.consrc ~~ '%_overview_constraint(%'::text)) AND (NOT pg_is_other_temp_schema(c.relnamespace)));
 
```

kalensk commented on ci/scripts/load-extensions.bash:
```diff
@@ -89,12 +89,6 @@ SQL_EOF
     # cluster due to deprication on 6X. Disabling ON_ERROR_STOP until this is
     # fixed.
     psql -v ON_ERROR_STOP=0 postgres -f /tmp/postgis_dump.sql
-    psql -v ON_ERROR_STOP=1 -d postgres <<SQL_EOF
-        -- Drop postgis views containing deprecated name datatypes
-        DROP VIEW geography_columns;
-        DROP VIEW raster_columns;
-        DROP VIEW raster_overviews;
-SQL_EOF
```
No longer dropping the postgis view's containing name data type results in the source and cluster dumps to be different.

Need to add a new filter for the following ARRAY ::text casting:

source 5X dump:
```SQL
AND ((c.relkind)::text = ANY ((ARRAY['r'::bpchar::character(1), 'v'::bpchar::character(1), 'm'::bpchar::character(1), 'f'::bpchar::character(1)])::text[]))) 
```

target 6X dump:
```SQL
AND ((c.relkind)::text = ANY (ARRAY[('r'::character(1))::text, ('v'::character(1))::text, ('m'::character(1))::text, ('f'::character(1))::text]))) 
```

More complete diff:
```diff
--- /tmp/source.sql.filtered	2023-03-10 03:15:36.532083771 +0000
+++ /tmp/target.sql.filtered	2023-03-10 03:15:21.043665813 +0000

 
 CREATE VIEW public.raster_columns AS
-SELECT current_database() AS r_table_catalog, n.nspname AS r_table_schema, c.relname AS r_table_name, a.attname AS r_raster_column, COALESCE(public._raster_constraint_info_srid(n.nspname, c.relname, a.attname), (SELECT public.st_srid('010100000000000000000000000000000000000000'::public.geometry) AS st_srid)) AS srid, public._raster_constraint_info_scale(n.nspname, c.relname, a.attname, 'x'::bpchar) AS scale_x, public._raster_constraint_info_scale(n.nspname, c.relname, a.attname, 'y'::bpchar) AS scale_y, public._raster_constraint_info_blocksize(n.nspname, c.relname, a.attname, 'width'::text) AS blocksize_x, public._raster_constraint_info_blocksize(n.nspname, c.relname, a.attname, 'height'::text) AS blocksize_y, COALESCE(public._raster_constraint_info_alignment(n.nspname, c.relname, a.attname), false) AS same_alignment, COALESCE(public._raster_constraint_info_regular_blocking(n.nspname, c.relname, a.attname), false) AS regular_blocking, public._raster_constraint_info_num_bands(n.nspname, c.relname, a.attname) AS num_bands, public._raster_constraint_info_pixel_types(n.nspname, c.relname, a.attname) AS pixel_types, public._raster_constraint_info_nodata_values(n.nspname, c.relname, a.attname) AS nodata_values, public._raster_constraint_info_out_db(n.nspname, c.relname, a.attname) AS out_db, public._raster_constraint_info_extent(n.nspname, c.relname, a.attname) AS extent FROM pg_class c, pg_attribute a, pg_type t, pg_namespace n WHERE (((((((t.typname = 'raster'::name) AND (a.attisdropped = false)) AND (a.atttypid = t.oid)) AND (a.attrelid = c.oid)) AND (c.relnamespace = n.oid)) AND ((c.relkind)::text = ANY ((ARRAY['r'::bpchar::character(1), 'v'::bpchar::character(1), 'm'::bpchar::character(1), 'f'::bpchar::character(1)])::text[]))) AND (NOT pg_is_other_temp_schema(c.relnamespace)));
+SELECT current_database() AS r_table_catalog, n.nspname AS r_table_schema, c.relname AS r_table_name, a.attname AS r_raster_column, COALESCE(public._raster_constraint_info_srid(n.nspname, c.relname, a.attname), (SELECT public.st_srid('010100000000000000000000000000000000000000'::public.geometry) AS st_srid)) AS srid, public._raster_constraint_info_scale(n.nspname, c.relname, a.attname, 'x'::bpchar) AS scale_x, public._raster_constraint_info_scale(n.nspname, c.relname, a.attname, 'y'::bpchar) AS scale_y, public._raster_constraint_info_blocksize(n.nspname, c.relname, a.attname, 'width'::text) AS blocksize_x, public._raster_constraint_info_blocksize(n.nspname, c.relname, a.attname, 'height'::text) AS blocksize_y, COALESCE(public._raster_constraint_info_alignment(n.nspname, c.relname, a.attname), false) AS same_alignment, COALESCE(public._raster_constraint_info_regular_blocking(n.nspname, c.relname, a.attname), false) AS regular_blocking, public._raster_constraint_info_num_bands(n.nspname, c.relname, a.attname) AS num_bands, public._raster_constraint_info_pixel_types(n.nspname, c.relname, a.attname) AS pixel_types, public._raster_constraint_info_nodata_values(n.nspname, c.relname, a.attname) AS nodata_values, public._raster_constraint_info_out_db(n.nspname, c.relname, a.attname) AS out_db, public._raster_constraint_info_extent(n.nspname, c.relname, a.attname) AS extent FROM pg_class c, pg_attribute a, pg_type t, pg_namespace n WHERE (((((((t.typname = 'raster'::name) AND (a.attisdropped = false)) AND (a.atttypid = t.oid)) AND (a.attrelid = c.oid)) AND (c.relnamespace = n.oid)) AND ((c.relkind)::text = ANY (ARRAY[('r'::character(1))::text, ('v'::character(1))::text, ('m'::character(1))::text, ('f'::character(1))::text]))) AND (NOT pg_is_other_temp_schema(c.relnamespace)));
 
 
 ALTER TABLE public.raster_columns OWNER TO gpadmin;
@@ -62068,7 +62069,7 @@
 --
 
 CREATE VIEW public.raster_overviews AS
-SELECT current_database() AS o_table_catalog, n.nspname AS o_table_schema, c.relname AS o_table_name, a.attname AS o_raster_column, current_database() AS r_table_catalog, (split_part(split_part(s.consrc, '''::name'::text, 1), ''''::text, 2))::name AS r_table_schema, (split_part(split_part(s.consrc, '''::name'::text, 2), ''''::text, 2))::name AS r_table_name, (split_part(split_part(s.consrc, '''::name'::text, 3), ''''::text, 2))::name AS r_raster_column, (btrim(split_part(s.consrc, ','::text, 2)))::integer AS overview_factor FROM pg_class c, pg_attribute a, pg_type t, pg_namespace n, pg_constraint s WHERE ((((((((((t.typname = 'raster'::name) AND (a.attisdropped = false)) AND (a.atttypid = t.oid)) AND (a.attrelid = c.oid)) AND (c.relnamespace = n.oid)) AND ((c.relkind)::text = ANY ((ARRAY['r'::bpchar::character(1), 'v'::bpchar::character(1), 'm'::bpchar::character(1), 'f'::bpchar::character(1)])::text[]))) AND (s.connamespace = n.oid)) AND (s.conrelid = c.oid)) AND (s.consrc ~~ '%_overview_constraint(%'::text)) AND (NOT pg_is_other_temp_schema(c.relnamespace)));
+SELECT current_database() AS o_table_catalog, n.nspname AS o_table_schema, c.relname AS o_table_name, a.attname AS o_raster_column, current_database() AS r_table_catalog, (split_part(split_part(s.consrc, '''::name'::text, 1), ''''::text, 2))::name AS r_table_schema, (split_part(split_part(s.consrc, '''::name'::text, 2), ''''::text, 2))::name AS r_table_name, (split_part(split_part(s.consrc, '''::name'::text, 3), ''''::text, 2))::name AS r_raster_column, (btrim(split_part(s.consrc, ','::text, 2)))::integer AS overview_factor FROM pg_class c, pg_attribute a, pg_type t, pg_namespace n, pg_constraint s WHERE ((((((((((t.typname = 'raster'::name) AND (a.attisdropped = false)) AND (a.atttypid = t.oid)) AND (a.attrelid = c.oid)) AND (c.relnamespace = n.oid)) AND ((c.relkind)::text = ANY (ARRAY[('r'::character(1))::text, ('v'::character(1))::text, ('m'::character(1))::text, ('f'::character(1))::text]))) AND (s.connamespace = n.oid)) AND (s.conrelid = c.oid)) AND (s.consrc ~~ '%_overview_constraint(%'::text)) AND (NOT pg_is_other_temp_schema(c.relnamespace)));
 
```

jimmyyih commented on test/acceptance/pg_upgrade/5-to-6/non_upgradeable_tests/sql/name_col.sql:
Since we're removing this file, we need to also remove the test entry from `non_upgradeable_schedule` as well.

jimmyyih reviewed (approved):
Looks good! 👍
